### PR TITLE
Fix timed domain resync

### PR DESCRIPTION
**Pull request by danielBelenky**
Fix logic issue with respect to Guest agent data. virt-launcher didn't always report data, and this was misinterpreted as the guest agent info being missing. 

**Release note**:
```release-note
Fix guest agent reporting.
```

### CODE REVIEW

File: pkg/virt-launcher/virtwrap/manager_test.go
Comment (dhiller) [inline pkg/virt-launcher/virtwrap/manager_test.go:427]: Please remove before merge.

**Comment by stu-gott** [comment]
/hold

There's one more unit test I think is behaving in an unexpected manner. This also probably manifests as a functional test failure. I'd like to review the new logical flow for post-copy migration a little bit more before calling this PR complete. That said, it's probably ready for critical review now.

/cc @rmohr @ashleyschuett

**Comment by stu-gott** [comment]
/hold cancel

After further investigation I believe the unit test I was concerned about is correct
/retest

**Comment by vladikr** [comment]
/approve

This looks good to me. 
I see that we don't test the issues this PR is fixing. However, I don't have a good suggestion on how to test it either...

File: pkg/virt-launcher/virtwrap/manager.go
Comment (davidvossel) [inline pkg/virt-launcher/virtwrap/manager.go:1432]: how did you determine these are the errors to check for?

**Review by davidvossel — COMMENTED** [review]
do we have any unit tests that validate the fail back to the inactive config?

File: pkg/virt-launcher/virtwrap/manager.go
Comment (vladikr) [reply pkg/virt-launcher/virtwrap/manager.go:1432]: that's what we do in https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-launcher/notify-client/client.go#L229

**Comment by stu-gott** [comment]
/hold cancel

After further investigation I believe the unit test I was concerned about is correct
/retest

> /approve
> 
> This looks good to me.
> I see that we don't test the issues this PR is fixing. However, I don't have a good suggestion on how to test it either...

Oh man! Good point. I was so focused on fixing all the tests this PR broke that I didn't think to investigate possible new tests. I can at least test the fallback to getting the domain without runtime info.

**Comment by vladikr** [comment]
thanks Stu! This looks good to me.
@davidvossel do you have any concerns?

**Review by davidvossel — APPROVED** [review]
/lgtm

> thanks Stu! This looks good to me.
> @davidvossel do you have any concerns?

i think we're good. we have unit test coverage, and i'd expect the existing func tests to uncover something if we introduced any sort of subtle issue here.

**Comment by rmohr** [comment]
@stu-gott great to see this in. :+1:

**Comment by stu-gott** [comment]
/cherry-pick release-0.34

Not a mistake to do this twice. This fix is needed on both these stable branches.

**Pull request merged**
